limitador/src/lib.rs

```diff
@@ -400,6 +400,39 @@ impl RateLimiter {
         }
     }
 
+    pub fn check_rate_limited_and_update_getting_counters(
+        &self,
+        namespace: &Namespace,
+        values: &HashMap<String, String>,
+        delta: i64,
+    ) -> Result<(bool, HashSet<Counter>), LimitadorError> {
+        let counters: HashSet<Counter> = self
+            .counters_that_apply(namespace, values)?
+            .into_iter()
+            .collect();
+
+        if counters.is_empty() {
+            self.prometheus_metrics.incr_authorized_calls(namespace);
+            return Ok((false, HashSet::new()));
+        }
+
+        let check_result = self.storage.check_and_update(counters.clone(), delta)?;
+
+        let counters = self.storage.load_counters(counters)?;
```

Comment:
This is quite a "big" window for races here. Data will be quite wrong and observable by the user for qualified counters... We should probably add a (hopefully race-free) way to have accurate counters. Also, worth considering the overhead... We currently do have a script [`VALUES_AND_TTLS`](https://github.com/Kuadrant/limitador/blob/main/limitador/src/storage/redis/scripts.rs#L26-L32) that does get the counter values and associated TTL in a single round trip to Redis. We should consider minimizing the overhead here... From slack:

Alex Snaps
>If that has say over 10% of degradation in latency, then I think we should make this optional in limitador's config...
>(kinda pulled this figure out of thin air tho)

Hiram Chirino
>+1 

So that the overhead of adding the headers is optional and can be (dis|en)abled by the user to...